### PR TITLE
Deployment changes

### PR DESCRIPTION
**Pull request by maltezacharias**
Umstellung auf neue Deploymentstruktur
Automatisches Deployment per Webhook

### CODE REVIEW

**Review by Baschdl — CHANGES_REQUESTED** [review]
Please change ./deploy.sh accordingly. I would use a parameter or an environment variable for dev/prod.

File: docker-compose.prod.yml [hunk 1]
```diff
@@ -3,28 +3,32 @@ version: '3.7'
...
      - DJANGO_SETTINGS_MODULE=match4healthcare.settings.production
+    user: ${CURRENT_UID:-0}
     command: gunicorn -c /match4healthcare-backend/gunicorn.conf.py match4healthcare.wsgi
 
   database:
+    container_name: ${BRANCH:-prod}-database
     image: postgres
     volumes:
-     - ./database/data:/var/lib/postgresql/data
+     - ./database/data:/database
```
Comment (Baschdl) [inline docker-compose.prod.yml:24]: What's the idea behind this?

File: mkdeployment-environment.sh [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+go get github.com/adnanh/webhook && echo -e "Webhook-Installation\t$OK" || { echo -e "Webhook-Installation\t$ERROR"; exit 1; }
+git clone -q -b "${PREDEPLOY_BRANCH}" --depth 1 "${REPO}" "${SERVERHOME}-predeploy" && echo -e "Git Clone ${PREDEPLOY_BRANCH} -> ${SERVERHOME}-predeploy\t$OK" || { echo -e "Git Clone\t$ERROR"; exit 1; }
+git clone -q -b "${DEPLOY_BRANCH}"    --depth 1 "${REPO}" "${SERVERHOME}-deploy"    && echo -e "Git Clone ${DEPLOY_BRANCH}    -> ${SERVERHOME}-deploy \t\t$OK" || { echo -e "Git Clone\t$ERROR"; exit 1; }
+
+echo "Creating Config files and folders"
+ln -s "${GOPATH}/bin/webhook" "${TOOLS}"
+mkdir -p "${TOOLS}/webhooks/github"
+cat > "${TOOLS}/webhooks/hooks.json" <<EOF
+[
```
Comment (Baschdl) [inline mkdeployment-environment.sh:52]: Could be placed in a separate file with parameters

**Comment by Baschdl** [comment]
Also concerning build on travis, do we wanna introduce a environment variable `STAGE` set to either `dev` or `prod`? We had another feature where it would be helpful.

**Comment by maltezacharias** [comment]
As discussed with @Baschdl closing this PR in favour of building and pulling Docker images, new issue will be created

**Review by Baschdl — COMMENTED** [review]
Salvageable:
- '${BACKEND_PORT:-80}:8000'
- non-root

**Pull request closed without merging**